helper/schema/schema.go

```diff
@@ -846,7 +956,11 @@ func (m schemaMap) internalValidate(topSchemaMap schemaMap, attrsOnly bool) erro
 			return fmt.Errorf("%s: Cannot be both Required and Computed", k)
 		}
 
-		if !v.Required && !v.Optional && !v.Computed {
+		if v.RequiredForImport {
+
+		} else if v.OptionalForImport {
+
+		} else if !v.Required && !v.Optional && !v.Computed {
```

Comment:
It would be nice to validate the identity schema separate from the other schemas, it would make logic like this a little cleaner.

Would this allow you to create invalid resource schemas that use `OptionalForImport` or `RequiredForImport`, and therefore have `Required`, `Optional`, and `Computed` as false?

Reply:
Technically, this `internalValidate()` should be schema type agnostic. I don't think it's even possible to pass information about the schema type to this method without some major refactoring. If we have resource identity schema-specific validation that might need to be a separate method that's called in `(*provider).InternalValidate()`. This is where the validation that blocks write-only attributes in non-managed resource schemas is: https://github.com/hashicorp/terraform-plugin-sdk/blob/19e5b30b5502fc58d3a7008f6ca5954aaef21717/helper/schema/provider.go#L235-L238

Reply:
We'll implement validation in a follow-up PR – I'll leave this discussion open for reference.